docs/using.md

```diff
@@ -183,6 +183,46 @@ See [the example PipelineRun](../examples/invocations/kritis-pipeline-run.yaml).
    that the `Task` needs to run.
 2. The `TaskRun` will also serve as a record of the history of the invocations of the
    `Task`.
+3. Another way of running a Task is embedding the TaskSpec in the taskRun yaml as shown in the following example
+
+```yaml
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: go-example-git
+spec:
+  type: git
+  params:
+  - name: url
+    value: https://github.com/pivotal-nader-ziada/gohelloworld
+---
+apiVersion: pipeline.knative.dev/v1alpha1
+kind: TaskRun
+metadata:
+  name: build-push-task-run-2
+spec:
+  trigger:
+    triggerRef:
+      type: manual
+  inputs:
+    resources:
+    - name: workspace
+      resourceRef:
+        name: go-example-git
+  taskSpec: 
+    inputs:
+      resources:
+      - name: workspace
+        type: git    
```

Comment:
This type is specified twice, once in the type of the `workspace` resource (line 194) and again here. Can one be removed? What happens if the resource is `type:foo` but I specify it as an input with `type:bar` ?